### ISSUE TITLE
Problem with update in this few days 

### ISSUE THREAD

**Issue by yu123569** [googlecolab/colabtools#3497]
I'm using colab pro to train my model with custom dataset on yolact edge. It's going well in March 6, 2023.
https://github.com/haotian-liu/yolact_edge

I got this error until yesterday (March 13 2023) :

 "Traceback (most recent call last):
  File "/content/gdrive/MyDrive/ColabNotebooks/yolact_edge-master/train.py", line 707, in <module>
    train(0, args=args)
  File "/content/gdrive/MyDrive/ColabNotebooks/yolact_edge-master/train.py", line 433, in train
    losses = backward_and_log(run_name, net_outs, targets, masks, num_crowds)
  File "/content/gdrive/MyDrive/ColabNotebooks/yolact_edge-master/train.py", line 325, in backward_and_log
    losses = criterion(out, targets, masks, num_crowds)
  File "/usr/local/lib/python3.9/dist-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/content/gdrive/MyDrive/ColabNotebooks/yolact_edge-master/yolact_edge/layers/modules/multibox_loss.py", line 159, in forward
    losses.update(self.lincomb_mask_loss(pos, idx_t, loc_data, mask_data, priors, proto_data, masks, gt_box_t, inst_data))
  File "/content/gdrive/MyDrive/ColabNotebooks/yolact_edge-master/yolact_edge/layers/modules/multibox_loss.py", line 442, in lincomb_mask_loss
    downsampled_masks = F.interpolate(masks[idx].unsqueeze(0), (mask_h, mask_w),
  **File "/usr/local/lib/python3.9/dist-packages/torch/nn/functional.py", line 3866, in interpolate
    raise ValueError(
ValueError: Input and output must have the same number of spatial dimensions, but got input with spatial dimensions of [256] and output size of (138, 138). Please provide input tensor in (N, C, d1, d2, ...,dK) format and output size in (o1, o2, ...,oK) format.**"

----------------------
I had try to downgrade:
CUDA 11.8 -> 11.6 (follow with correct version of cudnn and TensorRT)
Python 3.9-> 3.8
The tensorflow-gpu(2.9.2) and pytorch(1.13.1+cu116) version just as before. 
Total 4 combination of version, but still got the same error. 

If I start with Python 3.8, that mean pip will install newest version for all packages.
When numpy==1.24.2, this error will happen:

**"File "mtrand.pyx", line 920, in numpy.random.mtrand.RandomState.choice
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (6,) + inhomogeneous part"**

Downgrade numpy to 1.22.4 could be fix this. 

-------------------------------
I guess there have some python package get update to cause this situation, because I try to install same packages(No specify version for other package) still got the same error on windows 10.
### Any possibility? I need the pip list before Mar 6 to check which one go wrong, thanks.

**Comment by mayankmalik-colab** [comment]
Although I don't know what specific code are you referring from the repo mentioned above, the `ValueError` you are getting doesn't seem like a colab specific issue. I can see a lot of code in the repos is using very old versions of packages. I would encourage you to file it upstram and recommend using newer packages.

**Comment by yone7** [comment]
@yu123569 
I encountered the same error while running train.py in yolact++.
Solved wIth numpy 1.23.5 (previous version)!
With newer version of numpy 1.24.2 in my case, appear the erro avove.

**Comment by samsara-ku** [comment]
Just change like this, in the `yolact/utils/augmentations.py`

AS-IS
```
self.sample_options = (
            # using entire original input image
            None,
            # sample a patch s.t. MIN jaccard w/ obj in .1,.3,.4,.7,.9
            (0.1, None),
            (0.3, None),
            (0.7, None),
            (0.9, None),
            # randomly sample a patch
            (None, None),
        )
```

TO-BE
```
        self.sample_options = np.array(
            [
            # using entire original input image
            None,
            # sample a patch s.t. MIN jaccard w/ obj in .1,.3,.4,.7,.9
            (0.1, None),
            (0.3, None),
            (0.7, None),
            (0.9, None),
            # randomly sample a patch
            (None, None),
            ],
            dtype=object
        )
```

I already checked that the training code doesn't have any problem when I adopted the `numpy==1.23.5` as @yone7 said.

But when I used that version, I realized the problem was due to the deprecated operation in the latest numpy.

I already checked it is worked at the `numpy==1.26.2`!
